src/Ssh/Ssh/Common/SshBaseCmdlet.cs

```diff
@@ -398,6 +413,7 @@ protected internal void ValidateParameters()
 
         protected internal void SetResourceType()
         {
+            TargetMachineTags = new Dictionary<string, string>();
```

Comment:
Is there a way that we can separate the logic to set the Resource Type and the Tags? The common code is retrieving the generic resources in the resource group with the same name and supported resource type. The logic to set the port relies on the Generic resource retrieved. I'm thinking that we could have a method just for making the call to Azure to get the list of possible resources and return the result. Then we call SetResourceType with that as a parameter to so the logic to check if there are more than one supported resource and set the resource Type, and then we call a new method that retrieves the tag from the resource. What do you think?

Reply:
Maybe we can change SetResourceType() to not only set the resource type, but to also return the resource itself. Then we can pass that into the function that retrieves the tags.